lib/dry/configurable/compiler.rb

```diff
@@ -29,7 +29,9 @@ def visit_setting(node)
       # @api private
       def visit_nested(node)
         parent, children = node
-        visit(parent).nested(call(children))
+        name, opts = parent[1]
+
+        Setting.new(name, **opts, children: Settings.new(call(children)))
```

Comment:
To save ourselves from creating a throwaway object (i.e. a setting that we then have to recreate extend with the children), we build the `Setting` instance here directly rather than running its part of the AST back through the compiler.